File: content/manual/v8.1/source/includes/qe-tutorials/rust/range_query.rs
// start-enable-range
let encrypted_fields_map = doc! {
    "fields": [
        {
            "path":     "patientRecord.ssn",
            "bsonType": "string",
            "keyId":    Bson::Null,
            "queries": { "queryType": "equality" },
        },
        {
            "path":     "patientRecord.billing",
            "bsonType": "object",
            "keyId":    Bson::Null,
        },
        {
            "path":     "patientRecord.billAmount",
            "bsonType": "int",
            "queries": { "queryType": "range",
                          "sparsity": 1,
                          "min": 100,
                          "max": 2000, 
                          "trimFactor": 4 },
        },
    ]
};
// end-enable-range

// start-query-range
let query =
    doc! { "$and": [
        doc! { "patientRecord.billAmount": doc! { "$gt": 1000 } },
        doc! { "patientRecord.billAmount": doc! { "$lt": 2000 } }
    ]
};
let find_result = encrypted_coll.find_one(query).await?;

match find_result {
    Some(document) => println!("{:?}", document),
    None => println!("Document not found"),
}
// end-query-range

File: content/manual/v8.0/source/includes/qe-tutorials/ruby/range-query.rb
# frozen_string_literal: true

# This file contains code snippets demonstrating how to enable and use
# range queries with Queryable Encryption.

# start-enable-range
encrypted_fields_map = {
  encrypted_fields: {
    fields: [
      {
        path: "patientRecord.ssn",
        bsonType: "string",
        queries: { queryType: "equality" },
        keyId: nil
      },
      {
        path: "patientRecord.billing",
        bsonType: "object",
        keyId: nil
      },
      {
        path: "patientRecord.billAmount",
        bsonType: "int",
        queries: {
          queryType: "range",
          sparsity: 1,
          trimFactor: 4,
          min: 100,
          max: 2000
        },
        keyId: nil
      }
    ]
  }
}
# end-enable-range

# start-query-range
find_result = encrypted_collection.find(
  "patientRecord.billAmount" => { "$gt" => 1000, "$lt" => 2000 }
).first
puts find_result.inspect
# end-query-range
